proposals/lip-0061.md

```diff
@@ -517,14 +520,18 @@ An object following the schema `aggregateCommitSchema`, which can be added the n
 
 ```python
 def selectAggregateCommit() -> AggregateCommit:
-    # Note that the BFT parameters at height maxHeightCertified+1 are already authenticated by the previous certificate.
-    heightNextBFTParameters = getNextHeightBFTParameters(getBFTHeights().maxHeightCertified + 1)
-    if previous function call returns valid height:
-        nextHeight = min(heightNextBFTParameters-1, getBFTHeights().maxHeightPrecommitted)
-    elif previous function call returns Not Found error:
+    try:
+        # Note that the BFT parameters at height maxHeightCertified+1 are already authenticated by the previous certificate.
+        heightNextBFTParameters = getNextHeightBFTParameters(getBFTHeights().maxHeightCertified + 1)
+        nextHeight = min(heightNextBFTParameters -1, getBFTHeights().maxHeightPrecommitted)
+    except:  
+        # getNextHeightBFTParameters function call returns error, i.e., there are no new bft params since last certificate.
+        # attempt to create certificates for finalized blocks
         nextHeight = getBFTHeights().maxHeightPrecommitted
 
-    while nextHeight > getBFTHeights().maxHeightCertified:
+    nextHeight = max(nextHeight, MIN_CERTIFICATE_HEIGHT)    
```

Comment:
If I understand well, what you propose is equivalent to

```
heightNextBFTParameters = getNextHeightBFTParameters(getBFTHeights().maxHeightCertified+1)
nextHeight = min(max(heightNextBFTParameters -1, MIN_CERTIFICATE_HEIGHT), getBFTHeights().maxHeightPrecommitted)

```

which was in a previous version (see [this commit](https://github.com/LiskHQ/lips/pull/334/commits/0b50ca6141d3fe9791a5f270cac79f790ef213be)). Given the comments I received I got confused and reverted it. So I guess you suggest that this assignment was correct because it does not allow to create aggregate commits for non-finalized blocks. Is that right?

Reply:
Switched to the previous setting.